pkg/regionspan/span.go

```diff
@@ -180,3 +174,30 @@ func Intersect(lhs Span, rhs Span) (span Span, err error) {
 
 	return Span{Start: start, End: end}, nil
 }
+
+// IsSubSpan returns true if the sub span is parents spans
+func IsSubSpan(sub Span, parents ...Span) bool {
+	if bytes.Compare(sub.Start, sub.End) >= 0 {
+		log.Fatal("the sub span is invalid", zap.Reflect("sub span", sub))
+	}
+	for _, parent := range parents {
+		if StartCompare(parent.Start, sub.Start) <= 0 &&
+			EndCompare(sub.End, parent.End) <= 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// ComparableSpan returns a memcomparable span
+func ComparableSpan(span Span) Span {
+	return Span{
+		Start: codec.EncodeBytes(nil, span.Start),
+		End:   codec.EncodeBytes(nil, span.End),
+	}
+}
```

Comment:
In order to avoid confusion, what about define a new type span which encodes key in memcomparable format?